File: app/src/main/java/com/example/hsrrelicmanager/model/rules/action/EnhanceAction.java
package com.example.hsrrelicmanager.model.rules.action;

import com.example.hsrrelicmanager.Relic;

public class EnhanceAction implements Action {
    private int targetLevel;

    public EnhanceAction(int targetLevel) {
        this.targetLevel = targetLevel;
    }

    public int getTargetLevel() {
        return targetLevel;
    }

    public void setTargetLevel(int targetLevel) {
        this.targetLevel = targetLevel;
    }

    @Override
    public void apply(Relic relic) {
        // TODO: Do stuff
    }

    @Override
    public String toString() {
        return "Enhance (Lvl " + targetLevel + ")";
    }
}

File: app/src/main/java/com/example/hsrrelicmanager/model/rules/action/EnhanceAction.kt
package com.example.hsrrelicmanager.model.rules.action

import com.example.hsrrelicmanager.model.relics.Relic

class EnhanceAction(var targetLevel: Int) : Action() {
    override val name = "Enhance"
    override val image = Relic.Status.UPGRADE.image
    override fun apply(relic: Relic?): Relic {
        throw UnsupportedOperationException("Not implemented yet")
    }

    override fun toString(): String {
        return "Enhance (Lvl $targetLevel)"
    }
}